hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -463,16 +463,6 @@ object HoodieFileIndex extends Logging {
     if (listingModeOverride != null) {
       properties.setProperty(DataSourceReadOptions.FILE_INDEX_LISTING_MODE_OVERRIDE.key, listingModeOverride)
     }
-    val partitionColumns = metaClient.getTableConfig.getPartitionFields
```

Comment:
nice，and in our condition only merge this code would work well https://github.com/apache/hudi/pull/9862 then we would have more test for it